### PR TITLE
Add traded_on_exchanges screener

### PR DESCRIPTION
**Pull request by IvanIvanoff**
## Changes

```graphql
{
  allProjects {
    slug
    tradedOnExchanges # returns list of strings
    tradedOnExchangesCount # returns an integer
  }
}
```

```graphql
{
  allProjects(selector: {filters: [
    {
      name: "traded_on_exchanges"
      args: "{\"exchanges\":[\"LAToken\", \"Binance\"], \"exchanges_combinator\": \"and\"}"
    }
  ]}){
    slug
  }
}
```

```graphql
{
  allProjects(selector: {filters: [
    {
      name: "traded_on_exchanges"
      args: "{\"exchanges\":[\"LAToken\", \"Binance\"], \"exchanges_combinator\": \"or\"}"
    }
  ]}){
    slug
  }
}
```

```graphql
{
  getMarketExchanges {
    exchange
    assetsCount
    pairsCount
  }
}
```

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: priv/repo/migrations/20210609121501_create_asset_markets_table.exs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  def change do
+    create table(:asset_exchange_pairs) do
+      add(:base_asset, :string, null: false)
+      add(:quote_asset, :string, null: false)
+      add(:exchange, :string, null: false)
+      add(:source, :string, null: false)
+      add(:last_update, :utc_datetime)
+
+      timestamps()
```
Comment (tspenov) [inline priv/repo/migrations/20210609121501_create_asset_markets_table.exs:12]: Should't `exchange+base+quote` be unique key ?

File: lib/sanbase/scrapers/cryptocompare/cryptocompare_markets.ex [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      |> Enum.flat_map(fn {_key, list} ->
+        Enum.map(list, fn elem ->
+          Map.merge(elem, %{source: "cryptocompare", inserted_at: now, updated_at: now})
+        end)
+      end)
+
+    data
+    |> Enum.chunk_every(1000)
+    |> Enum.each(fn chunk ->
+      Sanbase.Repo.insert_all(Sanbase.Market, chunk, on_conflict: :nothing)
```
Comment (tspenov) [inline lib/sanbase/scrapers/cryptocompare/cryptocompare_markets.ex:46]: Shouldn't we update with last record on conflict ?

File: lib/sanbase/scrapers/cryptocompare/cryptocompare_markets.ex [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    end)
+  end
+
+  defp api_key(), do: Config.module_get(Sanbase.Cryptocompare, :api_key)
+
+  defp url() do
+    "https://min-api.cryptocompare.com/data/v2/pair/mapping/exchange" <>
+      "\?extraParams\=Santiment\&api_key\=#{api_key()}"
```
Comment (tspenov) [inline lib/sanbase/scrapers/cryptocompare/cryptocompare_markets.ex:54]: Let's user api key in header authentication

File: priv/repo/migrations/20210609121501_create_asset_markets_table.exs (hunk 1 above)
Comment (IvanIvanoff) [reply priv/repo/migrations/20210609121501_create_asset_markets_table.exs:12]: Yes, I totally forgot about that 😬

Comment (IvanIvanoff) [reply priv/repo/migrations/20210609121501_create_asset_markets_table.exs]: Added. I also included `source` in the unique key and added 2 indexes on `exchange` and `base_asset`

File: lib/sanbase/scrapers/cryptocompare/cryptocompare_markets.ex (hunk 2 above)
Comment (IvanIvanoff) [reply lib/sanbase/scrapers/cryptocompare/cryptocompare_markets.ex:41]: The unique key spans all the fields (I've removed the `last_updated` as it's some implementation details we don't use). There is now nothing to replace here, so I can leave :nothing

**Review by tspenov — APPROVED** [review]

**Pull request merged**